app/helpers/application_helper.rb

```diff
@@ -3,4 +3,8 @@ def full_title page_title = ""
     base_title = t "title_page"
     page_title.blank? ? base_title : "#{page_title} | #{base_title}"
   end
+
+  def init_cart
+    return session[:cart] = [] if session[:cart].nil?
```

Comment:
bỏ return ko cần thiết đi e
```suggestion
    session[:cart] = [] if session[:cart].nil?
```